pkg/internal/tracking/query_test.go

```diff
@@ -0,0 +1,61 @@
+package tracking
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/provider/resources"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestAppendMetadataToSql(t *testing.T) {
+	metadata := NewMetadata("123", resources.Account, CreateOperation)
+	sql := "SELECT 1"
+
+	bytes, err := json.Marshal(metadata)
+	require.NoError(t, err)
+
+	expectedSql := fmt.Sprintf("%s --%s %s", sql, MetadataPrefix, string(bytes))
+
+	newSql, err := AppendMetadataToSql(sql, metadata)
+	require.NoError(t, err)
+	require.Equal(t, expectedSql, newSql)
+}
+
+func TestParseMetadataFromSql(t *testing.T) {
+	metadata := NewMetadata("123", resources.Account, CreateOperation)
+	sql, err := AppendMetadataToSql("SELECT 1", metadata)
```

Comment:
Nit: this unit test depends on `AppendMetadataToSql`, but we can simply use an explicit query here.

Reply:
done